<?xml version="1.0"?>
<launch>
	<arg name="mode" default="$(env MOVEL_MODE)" doc="mode: [deployment, development]"/>

	<arg name="env_pack" default="$(find cob_default_env_config)" />
	<arg name="robot_env" default="$(optenv ROBOT_ENV !!NO_ROBOT_ENV_SET!!)"/>
	<arg name="robot_radius" default="0.50"/>
	<arg name="coverage_radius" default="0.50"/>
	<arg name="use_test_maps" default="true"/>

	<group if="$(eval mode == 'deployment')">
		<node pkg="ipa_room_exploration" type="room_exploration_client" name="room_exploration_client" output="screen">
			<param name="robot_env" value="$(arg robot_env)"/>
			<param name="use_test_maps" value="$(arg use_test_maps)"/>
			<param name="map_file" value="/home/movel/coverage-path-planning/src/map_files/test_image.png"/>
			<param name="file_points" value="/home/movel/coverage-path-planning/src/map_files/coordinates.txt"/>
			<rosparam command="load" file="$(find ipa_room_exploration)/ros/launch/room_exploration_client_params.yaml" />
			<rosparam command="load" file="$(env HOME)/.config/movel/config/coverage_path_planner/config/room_exploration_client_params.yaml" />
		</node>
	</group>

	<group unless="$(eval mode == 'deployment')">
		<node pkg="ipa_room_exploration" type="room_exploration_client" name="room_exploration_client" output="screen">
			<param name="robot_env" value="$(arg robot_env)"/>
			<param name="use_test_maps" value="$(arg use_test_maps)"/>
			<param name="map_file" value="/home/movel/coverage-path-planning/src/map_files/test_image.png"/>
			<param name="file_points" value="/home/movel/coverage-path-planning/src/map_files/coordinates.txt"/>
			<rosparam command="load" file="$(find ipa_room_exploration)/ros/launch/room_exploration_client_params.yaml" />
		</node>
	</group>


</launch>
